diff --git a/src/server/utils/__snapshots__/getIndexHash.test.ts.snap b/src/server/utils/__snapshots__/getIndexHash.test.ts.snap
index 8a26158..467df44 100644
--- a/lib/server/utils/__snapshots__/getIndexHash.test.ts.snap
+++ b/lib/server/utils/__snapshots__/getIndexHash.test.ts.snap
@@ -6,6 +6,6 @@ exports[`getIndexHash > getIndexHash({"hashed":true,"indexBlog":true,"blogDir":[
 
 exports[`getIndexHash > getIndexHash({"hashed":true,"indexDocs":true,"docsDir":["/does-not-exist/docs"]}) should return '1' 1`] = `null`;
 
-exports[`getIndexHash > getIndexHash({"hashed":true,"indexDocs":true,"docsDir":["/tmp/docs"]}) should return '0' 1`] = `"9de0582a"`;
+exports[`getIndexHash > getIndexHash({"hashed":true,"indexDocs":true,"docsDir":["/tmp/docs"]}) should return '0' 1`] = `"c9c4419d"`;
 
 exports[`getIndexHash > getIndexHash({"hashed":true,"indexDocs":true,"docsDir":["/tmp/index.js"]}) should return '1' 1`] = `null`;
diff --git a/src/server/utils/scanDocuments.ts b/src/server/utils/scanDocuments.ts
index 2890e79..ed3e362 100644
--- a/lib/server/utils/scanDocuments.js
+++ a/lib/server/utils/scanDocuments.js
@@ -42,23 +42,35 @@ export async function scanDocuments(
       });
 
       for (const section of sections) {
+        const slicedHash = getSlicedHash(section.hash, url);
+
         if (section.title !== pageTitle) {
+
+          if (typeof slicedHash === 'boolean' && slicedHash === false) {
+            continue;
+          }
+
           headingDocuments.push({
             i: getNextDocId(),
             t: section.title,
             u: url,
-            h: section.hash,
+            h: slicedHash,
             p: titleId,
           });
         }
 
         if (section.content) {
+
+          if (typeof slicedHash === 'boolean' && slicedHash === false) {
+            continue;
+          }
+
           contentDocuments.push({
             i: getNextDocId(),
             t: section.content,
             s: section.title || pageTitle,
             u: url,
-            h: section.hash,
+            h: slicedHash,
             p: titleId,
           });
         }
@@ -67,3 +79,13 @@ export async function scanDocuments(
   );
   return allDocuments;
 }
+
+function getSlicedHash(hash: string, url: string) {
+  if (hash.includes("#") && !hash.startsWith("#")) {
+    if (hash.startsWith(url) && hash[url.length] === "#") {
+      return hash.slice(url.length);
+    }
+    return false;
+  }
+  return hash;
+}
